kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -289,342 +290,274 @@ impl<'tcx> GotocCtx<'tcx> {
+            Intrinsic::SimdAnd => codegen_intrinsic_binop!(bitand),
             // TODO: `simd_rem` doesn't check for overflow cases for floating point operands.
             // <https://github.com/model-checking/kani/pull/2645>
-            "simd_div" | "simd_rem" => {
-                self.codegen_simd_div_with_overflow(fargs, intrinsic, place, loc)
+            Intrinsic::SimdDiv | Intrinsic::SimdRem => {
+                self.codegen_simd_div_with_overflow(fargs, intrinsic_str, place, loc)
             }
-            "simd_eq" => {
+            Intrinsic::SimdEq => {
                 self.codegen_simd_cmp(Expr::vector_eq, fargs, place, span, farg_types, ret_ty)
             }
-            "simd_extract" => {
+            Intrinsic::SimdExtract => {
                 self.codegen_intrinsic_simd_extract(fargs, place, farg_types, ret_ty, span)
             }
-            "simd_ge" => {
+            Intrinsic::SimdGe => {
                 self.codegen_simd_cmp(Expr::vector_ge, fargs, place, span, farg_types, ret_ty)
             }
-            "simd_gt" => {
+            Intrinsic::SimdGt => {
                 self.codegen_simd_cmp(Expr::vector_gt, fargs, place, span, farg_types, ret_ty)
             }
-            "simd_insert" => {
+            Intrinsic::SimdInsert => {
                 self.codegen_intrinsic_simd_insert(fargs, place, cbmc_ret_ty, farg_types, span, loc)
             }
-            "simd_le" => {
+            Intrinsic::SimdLe => {
                 self.codegen_simd_cmp(Expr::vector_le, fargs, place, span, farg_types, ret_ty)
             }
-            "simd_lt" => {
+            Intrinsic::SimdLt => {
                 self.codegen_simd_cmp(Expr::vector_lt, fargs, place, span, farg_types, ret_ty)
             }
-            "simd_mul" => self.codegen_simd_op_with_overflow(
+            Intrinsic::SimdMul => self.codegen_simd_op_with_overflow(
                 Expr::mul,
                 Expr::mul_overflow_p,
                 fargs,
-                intrinsic,
+                intrinsic_str,
                 place,
                 loc,
             ),
-            "simd_ne" => {
+            Intrinsic::SimdNe => {
                 self.codegen_simd_cmp(Expr::vector_neq, fargs, place, span, farg_types, ret_ty)
             }
-            "simd_or" => codegen_intrinsic_binop!(bitor),
-            "simd_shl" | "simd_shr" => {
-                self.codegen_simd_shift_with_distance_check(fargs, intrinsic, place, loc)
+            Intrinsic::SimdOr => codegen_intrinsic_binop!(bitor),
+            Intrinsic::SimdShl | Intrinsic::SimdShr => {
+                self.codegen_simd_shift_with_distance_check(fargs, intrinsic_str, place, loc)
+            }
+            Intrinsic::SimdShuffle(stripped) => {
+                assert!(fargs.len() == 3, "`simd_shuffle` had unexpected arguments {fargs:?}");
+                let n: u64 = self.simd_shuffle_length(stripped, farg_types, span);
```

Comment:
Why do this here? Wouldn't it be better to have `n` be the field for `Intrinsic::SimdShuffle`?

Reply:
Agreed. I simply copied this code from before the match and kept it as-is.